===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(recutl_write_db_to_file:writer)) at 3 (line 432, column 3) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/src/rec-writer.c
+++ b/table1/benchmarks/recutils-1.8/src/rec-writer.c
@@ -429,6 +429,7 @@ rec_write_record (rec_writer_t writer,
                    || (!writer->skip_comments_p && (num_elem != (num_elems - 1))))
             {
               if (!rec_writer_putc (writer, '\n'))
+                free(writer);
                 return false;
             }
 
